In [1]:
import pandas as pd

df = pd.read_excel('Task.xlsx')
df.head()

Loan ID  Bad Agreement current status Application branch name  \
0        1    0      Active / Terminated                branch#1   
1        2    1      Active / Terminated                branch#1   
2        3    1      Active / Terminated                branch#1   
3        4    1      Active / Terminated                branch#1   
4        5    1      Active / Terminated                branch#8   

  Application source Month (loan issued)  Month  Year Agreement dealer name  \
0             dealer             2017-04      4  2017             dealer#40   
1             dealer             2017-04      4  2017             dealer#40   
2             dealer             2017-04      4  2017             dealer#26   
3             dealer             2017-04      4  2017             dealer#51   
4             dealer             2017-04      4  2017             dealer#40   

  Agreement dealer region  ... Client monthly expenses  Source of income  \
0                region#7  ...                     0.0               NaN   
1                region#7  ...                     0.0               NaN   
2               region#14  ...                     0.0               NaN   
3                region#7  ...                     0.0               NaN   
4               region#13  ...                     0.0               NaN   

  Client region  Agreement with warrantor   Vehicle make  Fuel type  \
0      region#7                        no           Ford     diesel   
1     region#14                        no     Volkswagen     diesel   
2      region#7                        no     Volkswagen     petrol   
3      region#7                        no  Mercedes-benz     diesel   
4       no info                        no         Nissan     diesel   

   Transmission type  Body type  Manufacturing year  Mileage, km  
0             manual     saloon                2012       128270  
1             manual      other                2007       167408  
2             manual      coupe                2009       146113  
3          automatic  limousine                2005       194786  
4          automatic      wagon                2007       155990  

[5 rows x 47 columns]

In [2]:
##get all columns
df.columns

Index(['Loan ID', 'Bad', 'Agreement current status', 'Application branch name',
       'Application source', 'Month (loan issued)', 'Month', 'Year',
       'Agreement dealer name', 'Agreement dealer region', 'Assessment result',
       'Service version', 'First due date', 'Annual rate', 'Invoice day',
       'LTV rate', 'Loan amount (issued amount)', 'Principal', 'Interest',
       'Total payable amount', 'Monthly payment', 'Loan term', 'Term (unit)',
       'Incoming repayments total', 'Incoming repayments 1m',
       'Incoming repayments 3m', 'Incoming repayments 6m',
       'Incoming repayments 9m', 'Expected repayments total',
       'Expected repayments 1m', 'Expected repayments 3m',
       'Expected repayments 6m', 'Expected repayments 9m',
       'Client email domain', 'Client date of birth', 'Client gender',
       'Client monthly income', 'Client monthly expenses', 'Source of income',
       'Client region', 'Agreement with warrantor', 'Vehicle make',
       'Fuel type', 'Tran

In [3]:
##get datatypes of the columns
df.dtypes

Loan ID                                 int64
Bad                                     int64
Agreement current status               object
Application branch name                object
Application source                     object
Month (loan issued)                    object
Month                                   int64
Year                                    int64
Agreement dealer name                  object
Agreement dealer region                object
Assessment result                      object
Service version                         int64
First due date                 datetime64[ns]
Annual rate                           float64
Invoice day                             int64
LTV rate                              float64
Loan amount (issued amount)           float64
Principal                             float64
Interest                              float64
Total payable amount                  float64
Monthly payment                       float64
Loan term                         

In [5]:
### conver bad, month and year column to categorical column
df["Bad"] = pd.Categorical(df["Bad"])
df["Year"] = pd.Categorical(df["Year"])
df["Month"] = pd.Categorical(df["Month"])

In [6]:
##again check datatypes
df.dtypes

Loan ID                                 int64
Bad                                  category
Agreement current status               object
Application branch name                object
Application source                     object
Month (loan issued)                    object
Month                                category
Year                                 category
Agreement dealer name                  object
Agreement dealer region                object
Assessment result                      object
Service version                         int64
First due date                 datetime64[ns]
Annual rate                           float64
Invoice day                             int64
LTV rate                              float64
Loan amount (issued amount)           float64
Principal                             float64
Interest                              float64
Total payable amount                  float64
Monthly payment                       float64
Loan term                         

##### Inital Analysis

In [7]:
##check value of Bad column
df['Bad'].value_counts()

0    1735
1     441
Name: Bad, dtype: int64

There are 441 Loan applications that are considered Bad

In [8]:
##check Agreement current status column
df["Agreement current status"].value_counts()

Active / Terminated    1639
Early repaid            498
Paid                     39
Name: Agreement current status, dtype: int64

Out of 2176 loan applications 1639(i.e more than 75%) applications are still active and 537 are paid

In [9]:
# check description for LTV column
df["LTV rate"].describe()

count    2176.000000
mean        0.754370
std         0.119217
min         0.140000
25%         0.700000
50%         0.800000
75%         0.850000
max         0.880000
Name: LTV rate, dtype: float64

On an average LTV rate is 75.4%

Most of the finance companies keep 80% LTV ratio as threshold, anything below that is even better. If LTV rate rises above 80% borrowing costs can become higher, or borrowers may be denied loans 

In [10]:
## flag ltv rates
def ltv_flag(a):
    if a > 0.8:
        return 'Bad'
    else:
        return 'Good'

In [11]:
##creating a new column LTV Flag
df['LTV Flag'] = df['LTV rate'].apply(ltv_flag)
df.head()

Loan ID Bad Agreement current status Application branch name  \
0        1   0      Active / Terminated                branch#1   
1        2   1      Active / Terminated                branch#1   
2        3   1      Active / Terminated                branch#1   
3        4   1      Active / Terminated                branch#1   
4        5   1      Active / Terminated                branch#8   

  Application source Month (loan issued) Month  Year Agreement dealer name  \
0             dealer             2017-04     4  2017             dealer#40   
1             dealer             2017-04     4  2017             dealer#40   
2             dealer             2017-04     4  2017             dealer#26   
3             dealer             2017-04     4  2017             dealer#51   
4             dealer             2017-04     4  2017             dealer#40   

  Agreement dealer region  ... Source of income  Client region  \
0                region#7  ...              NaN       region#7   
1                region#7  ...              NaN      region#14   
2               region#14  ...              NaN       region#7   
3                region#7  ...              NaN       region#7   
4               region#13  ...              NaN        no info   

  Agreement with warrantor   Vehicle make  Fuel type  Transmission type  \
0                       no           Ford     diesel             manual   
1                       no     Volkswagen     diesel             manual   
2                       no     Volkswagen     petrol             manual   
3                       no  Mercedes-benz     diesel          automatic   
4                       no         Nissan     diesel          automatic   

   Body type  Manufacturing year  Mileage, km  LTV Flag  
0     saloon                2012       128270      Good  
1      other                2007       167408       Bad  
2      coupe                2009       146113       Bad  
3  limousine                2005       194786       Bad  
4      wagon                2007       155990       Bad  

[5 rows x 48 columns]

In [12]:
##check values of ltv flag column
df['LTV Flag'].value_counts()

Good    1314
Bad      862
Name: LTV Flag, dtype: int64

More than 39% customers paid less than 20% of the loan amount as down payment.

In [15]:
### check cusotmer where ltv flag is bad
df[['Loan amount (issued amount)','Interest','Total payable amount','Monthly payment','Incoming repayments total','Expected repayments total']][df['LTV Flag'] == 'Bad'].describe()

Loan amount (issued amount)      Interest  Total payable amount  \
count                   862.000000    862.000000            862.000000   
mean                  16608.021856  19907.176949          37340.836856   
std                    9260.458312  12590.424043          21392.783286   
min                    2550.000000    822.700000           6159.060000   
25%                    9777.747500  10740.367500          21111.650000   
50%                   13532.100000  16812.045000          31241.270000   
75%                   21671.750000  26776.300000          48742.645000   
max                   45000.000000  62916.310000         108061.310000   

       Monthly payment  Incoming repayments total  Expected repayments total  
count       862.000000                 855.000000                 862.000000  
mean        726.939954               12979.163275               37339.676763  
std         415.066629                9457.749221               21392.509820  
min         110.210000                   0.000000                6159.060000  
25%         438.080000                6423.515000               21111.650000  
50%         603.875000               10510.200000               31241.270000  
75%         909.615000               17298.200000               48742.645000  
max        4710.880000               62900.700000              108061.310000

Here we are able to get only 34.75% of the expected repayments.

In [16]:
df['Bad'][df['LTV Flag'] == 'Bad'].value_counts()

0    641
1    221
Name: Bad, dtype: int64

221 out of these 862 customers have agreements where it's considered Bad.

##### check records where agreements is Active/Terminated

In [17]:
### check records where agreements is Active/Terminated
df1 = df[df["Agreement current status"] == 'Active / Terminated']
df1.head()

Loan ID Bad Agreement current status Application branch name  \
0        1   0      Active / Terminated                branch#1   
1        2   1      Active / Terminated                branch#1   
2        3   1      Active / Terminated                branch#1   
3        4   1      Active / Terminated                branch#1   
4        5   1      Active / Terminated                branch#8   

  Application source Month (loan issued) Month  Year Agreement dealer name  \
0             dealer             2017-04     4  2017             dealer#40   
1             dealer             2017-04     4  2017             dealer#40   
2             dealer             2017-04     4  2017             dealer#26   
3             dealer             2017-04     4  2017             dealer#51   
4             dealer             2017-04     4  2017             dealer#40   

  Agreement dealer region  ... Source of income  Client region  \
0                region#7  ...              NaN       region#7   
1                region#7  ...              NaN      region#14   
2               region#14  ...              NaN       region#7   
3                region#7  ...              NaN       region#7   
4               region#13  ...              NaN        no info   

  Agreement with warrantor   Vehicle make  Fuel type  Transmission type  \
0                       no           Ford     diesel             manual   
1                       no     Volkswagen     diesel             manual   
2                       no     Volkswagen     petrol             manual   
3                       no  Mercedes-benz     diesel          automatic   
4                       no         Nissan     diesel          automatic   

   Body type  Manufacturing year  Mileage, km  LTV Flag  
0     saloon                2012       128270      Good  
1      other                2007       167408       Bad  
2      coupe                2009       146113       Bad  
3  limousine                2005       194786       Bad  
4      wagon                2007       155990       Bad  

[5 rows x 48 columns]

In [18]:
##check size of the dataframe
df1.shape

(1639, 48)

In [19]:
## check for any empty values
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1639 entries, 0 to 2175
Data columns (total 48 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   Loan ID                      1639 non-null   int64         
 1   Bad                          1639 non-null   category      
 2   Agreement current status     1639 non-null   object        
 3   Application branch name      1639 non-null   object        
 4   Application source           1639 non-null   object        
 5   Month (loan issued)          1639 non-null   object        
 6   Month                        1639 non-null   category      
 7   Year                         1639 non-null   category      
 8   Agreement dealer name        1453 non-null   object        
 9   Agreement dealer region      1453 non-null   object        
 10  Assessment result            1639 non-null   object        
 11  Service version              1639 non-null 

Here we can see more than 50% values are missing from Source of income column and few values are also missing from Incoming Repayments columns

In [20]:
##check discription of new dataset
df1[['Loan amount (issued amount)','Total payable amount','Monthly payment','Incoming repayments total','Expected repayments total']].describe()

Loan amount (issued amount)  Total payable amount  Monthly payment  \
count                  1639.000000           1639.000000      1639.000000   
mean                  16042.430146          37301.448865       702.388603   
std                    9213.811524          21497.792673       382.017371   
min                    2550.000000           6029.490000       110.210000   
25%                    9382.000000          21738.620000       433.280000   
50%                   13310.000000          31630.460000       600.390000   
75%                   20445.350000          47779.285000       868.165000   
max                   45000.000000         111502.700000      4015.180000   

       Incoming repayments total  Expected repayments total  
count                1621.000000                1639.000000  
mean                 9189.509729               37300.106583  
std                  6650.608657               21498.177000  
min                     0.000000                6029.490000  
25%                  4930.410000               21727.215000  
50%                  7383.670000               31630.460000  
75%                 11365.790000               47779.285000  
max                 50153.980000              111502.700000

1:- 18 records are missing from Incoming repayments total<br>
2:- Here we can clearly see that we are receving only 24.6% of the Expected repayments<br>
3:- Some records have Incoming repayments total = 0

In [21]:
## in the above table we can see min incoming repayments total = 0
## lets check these values
df1[['Loan ID','Bad','Agreement current status','Application branch name','Month (loan issued)','First due date','Incoming repayments total','Expected repayments total']][df1['Incoming repayments total'] == 0]

Loan ID Bad Agreement current status Application branch name  \
146       147   1      Active / Terminated               branch#13   
192       193   1      Active / Terminated                branch#8   
330       331   1      Active / Terminated                branch#1   
375       376   1      Active / Terminated                branch#4   
556       557   1      Active / Terminated               branch#10   
649       650   1      Active / Terminated               branch#10   
654       655   1      Active / Terminated               branch#10   
692       693   1      Active / Terminated                 no info   
888       889   1      Active / Terminated                 no info   
1064     1065   1      Active / Terminated               branch#10   
1154     1155   1      Active / Terminated               branch#13   
1181     1182   1      Active / Terminated                branch#8   
1440     1441   1      Active / Terminated               branch#10   
1555     1556   1      Active / Terminated                 no info   
1813     1814   1      Active / Terminated                branch#7   
2134     2135   1      Active / Terminated                branch#7   

     Month (loan issued) First due date  Incoming repayments total  \
146              2017-06     2017-06-20                        0.0   
192              2017-06     2017-07-02                        0.0   
330              2017-07     2017-08-23                        0.0   
375              2017-08     2017-08-20                        0.0   
556              2017-09     2017-10-28                        0.0   
649              2017-10     2017-11-15                        0.0   
654              2017-10     2017-11-16                        0.0   
692              2017-10     2017-11-15                        0.0   
888              2017-12     2017-12-20                        0.0   
1064             2018-01     2018-01-30                        0.0   
1154             2018-01     2018-02-16                        0.0   
1181             2018-01     2018-02-05                        0.0   
1440             2018-02     2018-03-25                        0.0   
1555             2018-03     2018-04-05                        0.0   
1813             2018-04     2018-05-05                        0.0   
2134             2018-06     2018-07-25                        0.0   

      Expected repayments total  
146                     9209.11  
192                    18567.42  
330                    22440.28  
375                    40017.65  
556                    21765.89  
649                    39322.29  
654                    43840.22  
692                    42601.44  
888                    16895.52  
1064                   54191.69  
1154                   21213.79  
1181                   18631.61  
1440                   42767.11  
1555                   11473.00  
1813                   20179.35  
2134                   40950.10

1:- Here for most of the customers First due date is after 1month so it makes sense if they haven't paid any EMI

In [22]:
#### Exclude above records from the dataframe
ind = df1[df1['Incoming repayments total'] == 0].index
df1.drop(ind, axis=0, inplace=True)

C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [24]:
df1.shape

(1623, 48)

In [25]:
### again analyze numerical data
df1[['Loan amount (issued amount)','Total payable amount','Monthly payment','Incoming repayments total','Expected repayments total']].describe()

Loan amount (issued amount)  Total payable amount  Monthly payment  \
count                  1623.000000           1623.000000      1623.000000   
mean                  16080.768953          37383.245977       704.167307   
std                    9234.781602          21546.296908       382.937329   
min                    2550.000000           6029.490000       110.210000   
25%                    9400.000000          21841.870000       433.985000   
50%                   13345.200000          31642.400000       600.470000   
75%                   20490.000000          47851.605000       870.095000   
max                   45000.000000         111502.700000      4015.180000   

       Incoming repayments total  Expected repayments total  
count                1605.000000                1623.000000  
mean                 9281.118548               37381.890462  
std                  6619.745922               21546.689302  
min                     0.710000                6029.490000  
25%                  5027.770000               21796.510000  
50%                  7460.350000               31642.400000  
75%                 11449.370000               47851.605000  
max                 50153.980000              111502.700000

1:- Still incoming repayments total is quite less than expected repayments total.<br>
2;- we have received only 24.8% of the expected repayments total.

In [30]:
df1['LTV rate'].describe()

count    1623.000000
mean        0.759618
std         0.108615
min         0.160000
25%         0.700000
50%         0.800000
75%         0.850000
max         0.880000
Name: LTV rate, dtype: float64

In [29]:
###check values for ltv_flag column 
df1['LTV Flag'].value_counts()

Good    996
Bad     627
Name: LTV Flag, dtype: int64

More than 38% customers have paid less than 20% amount as down payment.

In [31]:
df1['Bad'][df1['LTV Flag'] == 'Bad'].value_counts()

0    417
1    210
Name: Bad, dtype: int64

Out of these 627 customers 210 have agreement marked as Bad

In [32]:
df1["Bad"].value_counts()

0    1210
1     413
Name: Bad, dtype: int64

413 Active/Terminated applications are considered bad

In [33]:
## check records where bad = 1
df_bad = df1[["Loan ID","Bad","Application branch name","Incoming repayments total","Expected repayments total"]][df1["Bad"] == 1]
df_bad.describe()

Loan ID  Incoming repayments total  Expected repayments total
count   413.000000                 395.000000                 413.000000
mean    925.363196                6764.348987               37717.428329
std     616.022553                6071.454214               21779.624530
min       2.000000                   0.710000                6629.390000
25%     404.000000                2741.150000               22075.530000
50%     866.000000                4874.060000               30981.700000
75%    1414.000000                8650.065000               49105.210000
max    2170.000000               33581.620000              105811.260000

1:- 18 records are missing for Incoming repayments total where application is considered Bad<br>
2:- We are getting only 17.93% of the expected repayments total.

In [34]:
## create column Repayments % for the records where agreement is Bad
df_bad["Repayments%"] = (df["Incoming repayments total"] / df["Expected repayments total"])*100

In [35]:
### group the data by branches so that we can know about the performance of each branch
df_bad.groupby("Application branch name")["Incoming repayments total","Expected repayments total","Repayments%"].mean()

<ipython-input-35-295814aa57c7>:2: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df_bad.groupby("Application branch name")["Incoming repayments total","Expected repayments total","Repayments%"].mean()


Incoming repayments total  Expected repayments total  \
Application branch name                                                         
branch#1                               7770.452698               34865.309841   
branch#10                              8600.646056               42631.347568   
branch#11                               543.000000               42227.100000   
branch#12                              1961.326667               25778.273333   
branch#13                              5163.815581               30433.700682   
branch#2                               6747.322857               45839.670000   
branch#3                               2100.000000               22108.910000   
branch#4                               6642.073889               51862.787600   
branch#5                                950.000000               18194.220000   
branch#6                               3599.780000               45169.910000   
branch#7                               3128.766667               37873.515000   
branch#8                               6646.905238               33917.088636   
branch#9                               5198.000000               39694.767500   
no info                                4682.990886               31859.422250   

                         Repayments%  
Application branch name               
branch#1                   24.629151  
branch#10                  20.869917  
branch#11                   1.285904  
branch#12                   8.829278  
branch#13                  18.920846  
branch#2                   20.905356  
branch#3                    9.498433  
branch#4                   14.420411  
branch#5                    5.221438  
branch#6                    7.969420  
branch#7                    6.961451  
branch#8                   23.003529  
branch#9                   16.843475  
no info                    15.498401

1:- In the above table we have data for worst branch report(in terms of incoming repaymenrts total) where application is considered bad<br>
2:- We can ask the branches for a report why we are seeing such trends

In [36]:
### group data by years and then analyze numerical data
df1.groupby('Year')[['Incoming repayments total','Expected repayments total','Incoming repayments 1m','Expected repayments 1m','Incoming repayments 3m','Expected repayments 3m','Incoming repayments 6m','Expected repayments 6m','Incoming repayments 9m','Expected repayments 9m']].sum()

Incoming repayments total  Expected repayments total  \
Year                                                         
2017                 8179806.36                27689835.84   
2018                 6716388.91                32980972.38   

      Incoming repayments 1m  Expected repayments 1m  Incoming repayments 3m  \
Year                                                                           
2017               772813.54               982147.07              1744067.00   
2018              1070562.45              1148483.50              2242791.35   

      Expected repayments 3m  Incoming repayments 6m  Expected repayments 6m  \
Year                                                                           
2017              2024599.41              3204137.73              3588277.92   
2018              2391758.24              3996792.11              4256670.35   

      Incoming repayments 9m  Expected repayments 9m  
Year                                                  
2017              4587529.90              5151956.43  
2018              5665833.26              6121582.46

In [37]:
df["Interest"].describe()

count     2176.000000
mean     19941.406705
std      12867.924632
min        407.430000
25%      10846.202500
50%      16989.320000
75%      26260.625000
max      65302.700000
Name: Interest, dtype: float64

In [38]:
## check records where cutomers pay more than average Interest
df_mean_int = df[['Loan amount (issued amount)','Interest','Total payable amount','Monthly payment','Incoming repayments total','Expected repayments total']][df["Interest"] > 19941.406705]
df_mean_int

Loan amount (issued amount)  Interest  Total payable amount  \
0                        32900.00  39161.39              72561.39   
1                        21250.00  25008.33              46758.33   
2                        32650.00  38410.71              71560.71   
3                        19045.95  22801.74              42347.69   
4                        29848.00  35253.54              65601.54   
...                           ...       ...                   ...   
2165                     14602.00  21801.22              37603.22   
2166                     28200.00  41316.60              70716.60   
2169                     16590.00  25021.55              42811.55   
2170                     18500.00  27294.28              46994.28   
2174                     18960.00  28119.72              48279.72   

      Monthly payment  Incoming repayments total  Expected repayments total  
0             1206.84                   27908.31                   72561.39  
1              785.89                   17750.00                   46758.33  
2             1197.81                   27097.00                   71560.71  
3              706.25                   13958.53                   42347.69  
4             1096.56                    5294.00                   65601.54  
...               ...                        ...                        ...  
2165           633.48                    3950.00                   37603.22  
2166          1178.61                   38152.93                   70716.60  
2169           713.18                    2975.00                   42811.55  
2170           789.75                    5933.00                   46994.28  
2174           808.19                    7062.47                   48279.72  

[883 rows x 6 columns]

If Total Payable amount is calculated by adding Loan Amount and Interest then there are issues with the Total Payable amount column.

In [39]:
df_mean_int.describe()

Loan amount (issued amount)      Interest  Total payable amount  \
count                   883.000000    883.000000            883.000000   
mean                  24089.292265  32200.771971          57240.347361   
std                    8596.008356  10856.988089          19105.141452   
min                   12915.000000  19961.720000          34082.740000   
25%                   17439.260000  23720.510000          42505.660000   
50%                   21600.000000  28731.880000          51693.940000   
75%                   28589.275000  37778.870000          67129.415000   
max                   45000.000000  65302.700000         111502.700000   

       Monthly payment  Incoming repayments total  Expected repayments total  
count       883.000000                 872.000000                 883.000000  
mean        976.166750               17112.015218               57239.781110  
std         339.068752               11396.158998               19105.045785  
min         565.850000                   0.000000               34082.740000  
25%         709.495000                8788.430000               42505.660000  
50%         875.700000               13998.120000               51693.940000  
75%        1161.590000               22726.895000               67129.415000  
max        2138.320000               70614.610000              111502.700000

##### Analyze categorical data

In [42]:
df1.dtypes[df1.dtypes == 'object']

Agreement current status    object
Application branch name     object
Application source          object
Month (loan issued)         object
Agreement dealer name       object
Agreement dealer region     object
Assessment result           object
Term (unit)                 object
Client email domain         object
Client gender               object
Source of income            object
Client region               object
Agreement with warrantor    object
Vehicle make                object
Fuel type                   object
Transmission type           object
Body type                   object
Manufacturing year          object
LTV Flag                    object
dtype: object

In [43]:
## check gender value counts
df['Client gender'].value_counts()

m    1592
f     584
Name: Client gender, dtype: int64

In [44]:
df[['Bad','Client gender']].value_counts()

Bad  Client gender
0    m                1273
     f                 462
1    m                 319
     f                 122
dtype: int64

In [45]:
## check vechicle make
df['Vehicle make'].value_counts()

Volkswagen       562
Bmw              307
Opel             248
Ford             183
Audi             180
Mercedes-benz    109
Škoda             97
Renault           88
Mazda             70
Peugeot           53
Citroën           29
Seat              25
Dacia             22
Hyundai           21
Toyota            17
Fiat              16
Nissan            15
Volvo             15
Kia               13
Mitsubishi        12
Mini              11
Smart              9
Honda              9
Alfa Romeo         9
Land Rover         9
Chevrolet          7
Jaguar             6
Chrysler           5
Saab               5
Suzuki             5
Lexus              4
Peugot             3
Jeep               3
Subaru             2
Rover              2
Porsche            1
Lancia             1
Dodge              1
Ssangyong          1
Renaul             1
Name: Vehicle make, dtype: int64

In [46]:
df['Body type'].value_counts()

sedan                      480
other                      453
hatchback                  244
wagon                      221
suv                         83
estate                      83
sedan-limousine             62
universal                   52
break                       46
no info                     36
mpv                         36
saloon                      35
monospace                   34
estate-hatchback            33
coupe                       32
crossover                   22
minivan                     21
tourer                      20
enduro                      20
limousine                   18
Wagon                       14
furgon                      11
Hatchback                   11
Sedan                       11
hatch                        9
cabrio                       6
sport-coupe                  6
hatchback-5d                 6
5-door-hatch                 5
off-road                     5
touring                      5
van                          4
5-door-e

In [47]:
df['Manufacturing year'].value_counts()

2006       320
2005       299
2007       252
2004       205
2001       170
2002       164
2003       162
2008       148
2009       118
2010        71
2000        69
2011        62
2012        37
1999        21
1998        19
2013        18
2014        15
1997         9
2017         5
no info      5
1996         3
2016         2
2015         1
1988         1
Name: Manufacturing year, dtype: int64